docs/DeveloperGuide.md

```diff
@@ -1,34 +1,383 @@
+using generateScreen function, which writes the intended content into
+the 2-D array of characters. After that, the array will be shown to the user
+by Ui class via the OutputStream. 
+
+Moreover, WeekStructure and MonthStructure extends the DisplayDateStructure.
+These classes will generate a different size of the 2-D array mentioned
+in the previous paragraph due the difference in number of days displayed.
+
+### Logic component
+
+
+![LogicComponent](diagrams/LogicClass.png)
+
+
+From the Ui component, the raw input from the User is passed to the Parser class.
+Parser class will process and validate the input, then it generates a specific
+command accordingly.  Specific commands extends from the abstract Command class.
+Command operates on the TaskMap and execution of Command generates a 
+CommandResult. This CommandResult is passed back to the Ui component and 
+respective content will be displayed to User based on the CommandResult. 
+
+### Model component
+
+
+![ModelComponent](diagrams/TaskMapClass.png)
+
+
+Task class consists of attributes like description, date, start time, end time,
+priority that are important to users. Whereas other attributes like 
+HASH_VALUE_DIGITS, taskID and internal values of Priority enum class matters
+more to developers. For example, the HASH_VALUE_DIGITS and the color code of 
+Priority could affect the content displayed to User, and the reason why the
+constant 4 is used will be explained in the next paragraph. 
+
+TaskMap is the container used for runtime storage, it stores up to 10000 Task
+objects. The number 10000 is chosen such that the unique identifier of each
+Task is kept within 4 digits. Keeping it to 4 digits allows the app to have a
+better displaying effect.
+
+The underlying data structure used for TaskMap is the Java LinkedHashMap.
+It is chosen because it allows efficient access to an element (Task) using the
+unique identifier (Integer). Another reason of using LinkedHashMap is because 
+it is ordered, this allows the container to be sorted. DateSorter and 
+PrioritySorter implements the Task comparator, they are used in the TaskMap
+functions sortByDate() and sortByPriority().
+
+
+### Storage component
+
+
+![StorageComponent](diagrams/StorageClass.png)
+
+
+At the initiating and closing stage of the program, the Storage component
+will read and write to the data file respectively. As shown in the diagram
+above, the data file is named as "data.json" and placed under a folder called
+"data". The Storage component also uses the Google Gson library to convert Task
+object to and from json format. For example, in the process of saving to file,
+each Task object is convert to json string and then writes to file via FileWriter. 
+
+
+### Common classes
+
+Common classes provide utility functions and string messages used by multiple components,
+and these classes are stored in the seedu.commons package.
 
 
 ## Product scope
```

Comment:
Wouldn't it better if you guys can move the product scope to the front of the document to introduce the reader to the app first, rather than starting off with the complex diagrams?